### PR TITLE
Contribution controller

### PR DESCRIPTION
**Pull request by lepetitbelge**
created basic logic for contribution controller -> need view of speech to continue working on it

### CODE REVIEW

File: app/controllers/contributions_controller.rb [hunk 1]
```diff
@@ -0,0 +1,46 @@
+class ContributionsController < ApplicationController
+  before_action :find_contribution, only: %i[update]
+
+  def create
+    # maybe need to make a speech here, can't test this fully until we have a speech view
```
Comment (ghost) [inline app/controllers/contributions_controller.rb:5]: nope, I don't think so

File: app/controllers/contributions_controller.rb [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    @contribution = Contribution.new(contribution_params)
```
Comment (ghost) [inline app/controllers/contributions_controller.rb]: no '@', we don't send it to the view

File: app/controllers/contributions_controller.rb [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    @contribution.speech = find_speech
+    if @contribution.save
+      puts "Save went well, we might render with AJAX"
+    else
+      puts "Save went awfully wrong"
+    end
+  end
+
+  def update
+    @contribution.update(contribution_update_params)
```
Comment (ghost) [inline app/controllers/contributions_controller.rb]: don't need this line (17)

File: app/controllers/contributions_controller.rb [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    if @contribution.save
```
Comment (ghost) [inline app/controllers/contributions_controller.rb]: instead:
if contribution.update(contribution_update_params)

File: app/controllers/contributions_controller.rb [hunk 5]
```diff
@@ -0,0 +1,46 @@
+class ContributionsController < ApplicationController
+  before_action :find_contribution, only: %i[update]
```
Comment (ghost) [inline app/controllers/contributions_controller.rb]: %i[update destroy]

File: app/controllers/contributions_controller.rb [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+  def contribution_params
+    params.require(:contribution).permit(:content, :start_position, :end_position)
+  end
+
+  def contribution_update_params
+    params.require(:contribution).permit(:content)
+  end
+
+  def find_contribution
+    @contribution = Contribution.find(params[:contribution_id])
```
Comment (ghost) [inline app/controllers/contributions_controller.rb]: don't need '@'

File: app/controllers/contributions_controller.rb [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+  end
+
+  def find_speech
+    @speech = Speech.find(params[:speech_id])
```
Comment (ghost) [inline app/controllers/contributions_controller.rb]: don't need the '@'

File: app/controllers/contributions_controller.rb (hunk 5 above)
Comment (ghost) [inline app/controllers/contributions_controller.rb]: skip_before_action ...

Comment (lepetitbelge) [reply app/controllers/contributions_controller.rb]: Check!

Comment (lepetitbelge) [reply app/controllers/contributions_controller.rb]: not in this case, as we need user_id for contributions

**Comment by ghost** [comment]
disagree

**Pull request merged**